src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -74,6 +77,7 @@ export const RequestPage = () => {
   const [isTransactionModalOpen, setIsTransactionModalOpen] = useState(false)
   const requestRef = useRef<RecoverResponse>()
   const [error, setError] = useState<string>()
+  const [ipValidationError, setIpValidationError] = useState<{ requestId: string; reason: string }>()
```

Comment:
As the `requestId` can be retrieved from the `requestId` in line 82, WDYT about not defining a new state and using the current `error` state to define the reason?

Reply:
Refactored, thanks!